tests/test_loky_module.py

```diff
@@ -64,3 +67,39 @@ def test_cpu_count_cfs_limit():
                         'python', '-c', cpu_count_cmd])
 
     assert res.strip().decode('utf-8') == '1'
+
+
+@pytest.mark.parametrize("clib", _CLibsWrapper.SUPPORTED_CLIBS)
+def test_limit_threads_clib(openblas_test_noskip, mkl_win32_test_noskip, clib):
+    thread_limits = get_thread_limits()
+
+    old_thread_limit = thread_limits[clib]
+    if old_thread_limit is None:
+        if clib == "openblas" and openblas_test_noskip:
+            raise RuntimeError("Could not load the OpenBLAS library")
+        elif clib == "mkl_win32" and mkl_win32_test_noskip:
+            raise RuntimeError("Could not load the MKL library")
+        pytest.skip("Need {} support".format(clib))
+
+    dynamic_scaling = limit_threads_clib(1)
+    assert get_thread_limits()[clib] == 1
+    assert dynamic_scaling[clib]
+
+    limit_threads_clib(3)
+    max_threads = get_thread_limits()[clib]
+    assert max_threads == 3 or max_threads == min(3, os.cpu_count())
+
```

Comment:
I suspect this won't pass with mkl 2018, since it's limited by the number of physical core, not virtual cpus.